### PR TITLE
Bump ruff and add new rules

### PR DESCRIPTION
**Pull request by harshil21**
wake up, new ruff just dropped! 

new rules added:

- print: https://github.com/python-telegram-bot/python-telegram-bot/pull/4311#discussion_r1657902724
- "furb": https://docs.astral.sh/ruff/rules/#refurb-furb

### CODE REVIEW

**Review by Bibo-Joshi — APPROVED** [review]

**Review by Bibo-Joshi — CHANGES_REQUESTED** [review]
Upsie, no, pre-commit is failing I should not approve that :D

**Comment by harshil21** [comment]
> Upsie, no, pre-commit is failing I should not approve that :D

the ruff fix for job queue is probably going to break the type completeness again, do you want to ignore it?

File: examples/passportbot.py [hunk 1]
```diff
@@ -47,9 +47,9 @@
     # Files will be downloaded to current directory
     for data in passport_data.decrypted_data:  # This is where the data gets decrypted
         if data.type == "phone_number":
-            print("Phone: ", data.phone_number)
+            logger.info("Phone: %s", data.phone_number)
```
[the remark this answers is not in the document]
Comment (Bibo-Joshi) [reply examples/passportbot.py:50]: Huh, first time I see this security feature doing something:D

**Review by Bibo-Joshi — APPROVED** [review]
Anything left from your side?

**Comment by harshil21** [comment]
all good

**Pull request merged**